apps/consent-next.js/app/logout/page.tsx

```
@@ -0,0 +1,76 @@
+import { redirect } from "next/navigation";
+import React from "react";
+import { hydraClient } from "../hydra-config";
+
+interface logOutProps {
+  logout_challenge: string;
+}
+
+const submitForm = async (form: FormData) => {
+  "use server";
+  const logout_challenge = form.get("logout_challenge");
+  const submitValue = form.get("submit");
+
+  if (
+    typeof logout_challenge === "string" &&
+    typeof submitValue === "string" &&
+    logout_challenge &&
+    submitValue
+  ) {
+    if (submitValue === "No") {
+      await hydraClient.rejectOAuth2LogoutRequest({
+        logoutChallenge: logout_challenge,
+      });
+      redirect("https://www.blink.sh/");
+    }
+    const response = await hydraClient.acceptOAuth2LogoutRequest({
+      logoutChallenge: logout_challenge,
+    });
+    redirect(String(response.data.redirect_to));
+  }
+};
```

Should return an error otherwise